holoviews/plotting/bokeh/element.py

```diff
@@ -508,11 +508,14 @@ def _axis_props(self, plots, subplots, element, ranges, pos, *, dim=None,
                 l, b, r, t = b, l, t, r
             if pos == 1 and self._subcoord_overlaid:
                 if isinstance(self.subcoordinate_y, bool):
-                    offset = self.subcoordinate_scale / 2.
-                    # This sum() is equal to n+1, n being the number of elements contained
-                    # in the overlay with subcoordinate_y=True, as the traversal goes through
-                    # the root overlay that has subcoordinate_y=True too since it's propagated.
-                    v0, v1 = 0-offset, sum(self.traverse(lambda p: p.subcoordinate_y))-2+offset
+                    if self.ylim and all(np.isfinite(val) for val in self.ylim):
+                        v0, v1 = self.ylim
+                    else:
+                        offset = self.subcoordinate_scale / 2.
+                        # This sum() is equal to n+1, n being the number of elements contained
+                        # in the overlay with subcoordinate_y=True, as the traversal goes through
+                        # the root overlay that has subcoordinate_y=True too since it's propagated.
```

Comment:
Took me some time to parse this originally

```suggestion
                        # This sum() is equal to n+1, where n is the number of elements contained
                        # in the overlay with subcoordinate_y=True (including the the root overlay,
                        # which has subcoordinate_y=True due to option propagation)
```